test/functional/android-helper-e2e-specs.js

```diff
@@ -54,4 +52,19 @@ describe('android-helpers e2e', () => {
       }
     });
   });
+  describe('pushSettingsApp', function () {
+    const settingsPkg = 'io.appium.settings';
+    it('should be able to upgrade from settings v1 to latest', async function () {
+      await adb.uninstallApk(settingsPkg);
+
+      // get and install old version of settings app
+      await exec('npm', ['install', `${settingsPkg}@2.0.0`]);
+      await helpers.pushSettingsApp(adb, true).should.eventually.be.rejected;
```

Comment:
I thought that _eventually_ makes _await_ redundant, isn't it?

Reply:
Kind of. But without the `await` the execution would just go on to the next command without waiting.

Reply:
Essentialy `eventually` just puts a `catch` on the promise that is returned, with an assertion inside that that block gets run. We need to `await` still so we don't move on until the promise is fulfilled (hopefully rejected).